typescript/sdk/src/app/MultiProtocolApp.ts

```diff
@@ -103,4 +117,16 @@ export abstract class MultiProtocolApp<
   ): Promise<ChainMap<Output>> {
     return promiseObjAll(objMap(this.adapters(), fn));
   }
+
+  metadata(chain: ChainName): ChainMetadata<ContractAddrs> {
```

Comment:
Exactly, the config refactors lead to this.

At first, I had that concept across the adapters too but it felt too much like the old strict chain typing we used to have across the SDK, so I pulled it back to just MultiProtocolApps in case someone wants to use adapters directly. 

I'm not 100% sure this is the way to go. If you're instinct is that we'd be better keeping artifacts + metadata separate in Apps too, I'm cool with that

Reply:
Updated